src/libs/ck-libs/ampi/ampi.C

```diff
@@ -2213,6 +2216,24 @@ CProxy_ampi ampi::createNewChildAmpiSync() noexcept {
   return newAmpi;
 }
 
+CProxy_ampi ampi::createNewSplitCommArray(MPI_Comm newComm, const std::vector<int> & indices) noexcept
+{
+  CProxy_ampi lastAmpi = createNewChildAmpiSync();
+
+  //FIXME: create a new communicator for each color, instead of
+  // (confusingly) re-using the same MPI_Comm number for each.
+  ampiCommStruct lastComm = ampiCommStruct(newComm, lastAmpi, indices);
+
+  for (int newIdx : indices)
+  {
+    lastAmpi[newIdx].insert(parentProxy, lastComm);
+  }
+
+  lastAmpi.doneInserting();
```

Comment:
Cn you add a comment here that control will be returned via the initCallback to ampi::registrationFinish() after doneInserting() runs